### PR TITLE
Introduction of number_is_zero in C wrappers

### LINKED ISSUE #1254: Introduction of "number_is_zero" in C wrappers?
Hi,

I am asking, whether it would be a good idea to replace the following two functions

```
int real_mpfr_is_zero(const basic s);
int complex_mpc_is_zero(const basic s);
```
by just the following one?
```
int number_is_zero(const basic s)
{
    SYMENGINE_ASSERT(is_a_Number(*(s->m)));
    return (int)((down_cast<const Number &>(*(s->m))).is_zero());
}
```

This would allow to check all numbers with the same function for being zero or not. Do you think
this is a good idea, or are there any drawbacks?

Thank you,

Dirk

### PR DESCRIPTION
**Pull request by rdbyk**
Ok, I finally managed to make this PR. Is everything Ok?

If yes, what about having `number_is_negative`, `number_is_positive`, `number_is_complex`?

I could try to do that as well.

Fixes #1254

### CODE REVIEW

File: symengine/cwrapper.cpp [hunk 1]
```diff
@@ -232,6 +232,12 @@ CWRAPPER_OUTPUT_TYPE symbol_set(basic s, const char *c)
     CWRAPPER_END
 }
 
+int number_is_zero(const basic s)
+{
+	SYMENGINE_ASSERT(is_a_Number(*(s->m)));
+	return (int)((down_cast<const Number &>(*(s->m))).is_zero());
```
Comment (isuruf) [inline symengine/cwrapper.cpp]: Can you change the two tabs starting the two lines above with 4 four space each?

**Comment by rdbyk** [comment]
I haven't done it yet and would prefer to do it in a separate PR, if that's Ok with you?

Do you agree, that `basic_number_sign` should be dropped, when `number_is_negative` and `number_is_positive` is available?

**Comment by isuruf** [comment]
Ah, I totally forgot that `basic_number_sign` is there. Maybe then we only need `basic_number_is_complex`?

**Comment by rdbyk** [comment]
I think it's cleaner to drop `basic_number_sign`, because it introduces an implementation of a new `sign` functionality and it is not just a wrapper as i would expect it to be.

With `number_is_xxx` one can easily build your own 'sign' functionality as needed and it is totally clear what is wrapped here.

What do you think?

**Comment by isuruf** [comment]
@rdbyk, the other PR caused conflicts. Can you resolve the conflicts? You can do it on the web editor by clicking the button `Resolve conflicts` below.

**Comment by rdbyk** [comment]
@isuruf I will try do it, so I am learning one thing more regarding git(hub).

**Comment by isuruf** [comment]
You can also delete `complex_mpc_is_zero`

**Comment by rdbyk** [comment]
I hope I didn't spoil the format, naturally I have forgotten to use clang-format before pushing ... we will see.

**Comment by rdbyk** [comment]
ping @isuruf

**Pull request merged**